### PR TITLE
Add Storj  DCS connection profile

### PR DESCRIPTION
**Pull request by wthorp**
Create a profile for the Storj DCS network's main Gateway-MT URL.  This is an S3-compatible endpoint.

### CODE REVIEW

File: Storj.cyberduckprofile [hunk 1]
```diff
@@ -0,0 +1,2558 @@
...
+<key>Protocol</key>
+<string>s3</string>
+<key>Vendor</key>
+<string>Storj Labs</string>
+<key>Description</key>
+<string>Storj DCS</string>
+<key>Default Hostname</key>
+<string>gateway.storjshare.io</string>
+<key>Hostname Configurable</key>
+<true/>
```
Comment (dkocher) [inline Storj.cyberduckprofile:12]: Are you sure you want to allow other endpoints than `gateway.storjshare.io`?

File: Storj.cyberduckprofile [hunk 2]
```diff
@@ -0,0 +1,2558 @@
...
+<string>AWS4HMACSHA256</string>
+<key>Properties</key>
+<array>
+<string>s3.storage.class.options=STANDARD</string>
+<string>s3.upload.multipart.size=67108864</string>
+<string>s3.copy.multipart.size=67108864</string>
+<string>s3.versioning.enable=false</string>
+<string>s3.accelerate.enable=false</string>
+<string>s3.bucket.requesterpays=false</string>
+</array>
```
Comment (dkocher) [inline Storj.cyberduckprofile:31]: You may also want to restrict the storage class options using `s3.storage.class.options`.

**Review by dkocher — CHANGES_REQUESTED** [review]
Please also specify available regions using 

```
        <key>Region</key>
        <string>us-east-1</string>
        <key>Regions</key>
        <array>
            <string>us-east-1</string>
        </array>
```

Can you pretty print the XML for better readability.

File: Storj.cyberduckprofile (hunk 2 above)
Comment (wthorp) [reply Storj.cyberduckprofile:32]: Done.  I suspect this doesn't matter on our side, in which case adding it won't hurt.

File: Storj.cyberduckprofile (hunk 1 above)
Comment (wthorp) [reply Storj.cyberduckprofile:12]: I think so.  Its possible to connect directly to regional PoP endpoints such as https://gateway.us1.storjshare.io/.  While there is seldom any utility in doing so, it seems sensible to allow this.

**Comment by wthorp** [comment]
> 
Pretty printing is complete.

I'm curious if the request to add regions applies to Storj.  We don't have the concept of regions in our network, and our back-end logic should ignore client-supplied regions.  More precisely, Storj's S3 compatibility assumes an empty string region by default, but will respect client-supplied region strings when verifying secret key signatures.

**Comment by dkocher** [comment]
> I'm curious if the request to add regions applies to Storj. We don't have the concept of regions in our network, and our back-end logic should ignore client-supplied regions. More precisely, Storj's S3 compatibility assumes an empty string region by default, but will respect client-supplied region strings when verifying secret key signatures.

The `Region` in the profile will determine the default region to be used when signing requests when we cannot determine the location of the bucket using the `?location` query. The `Regions` in the profile will determine the available regions to choose from when creating a new bucket.

**Review by dkocher — DISMISSED** [review]

**Comment by dkocher** [comment]
4742f048509c7a2c338656b5903e064d92b94be8 seems to accidentally include the template image.

**Comment by dkocher** [comment]
Please sqash the commits.

**Review by dkocher — APPROVED** [review]

**Pull request merged**